src/parquet/thrift/util.h

```diff
@@ -34,13 +35,44 @@ inline void DeserializeThriftMsg(const uint8_t* buf, uint32_t* len, T* deseriali
       tproto_factory.getProtocol(tmem_transport);
   try {
     deserialized_msg->read(tproto.get());
-  } catch (apache::thrift::protocol::TProtocolException& e) {
-    throw ParquetException("Couldn't deserialize thrift.", e);
+  } catch (std::exception& e) {
+    //TODO Determine an appproach to log exceptions
+    LOG(1) << "Couldn't deserialize thrift: " << e.what();
+    return false;
   }
   uint32_t bytes_left = tmem_transport->available_read();
   *len = *len - bytes_left;
+  return true;
 }
 
+// Serialize obj into a memory buffer.  The result is returned in buffer/len.  The
+// memory returned is owned by this object and will be invalid when another object
+// is serialized.
+template <class T>
+inline bool SerializeThriftMsg(T* obj, uint32_t* len, std::vector<uint8_t>* result) {
+  uint8_t* buf = NULL;
+  boost::shared_ptr<apache::thrift::transport::TMemoryBuffer> mem_buffer(
+      new apache::thrift::transport::TMemoryBuffer(*len));
+  apache::thrift::protocol::TCompactProtocolFactoryT<
+      apache::thrift::transport::TMemoryBuffer> tproto_factory;
+  boost::shared_ptr<apache::thrift::protocol::TProtocol> tproto =
+      tproto_factory.getProtocol(mem_buffer);
+  try {
+    mem_buffer->resetBuffer();
+    obj->write(tproto.get());
+  } catch (std::exception& e) {
+    std::stringstream msg;
+    LOG(1) << "Couldn't serialize thrift object: " << e.what();
+    return false;
+  }
+  mem_buffer->getBuffer(&buf, len);
+  result->resize(*len);
+  memcpy(&((*result)[0]), buf, *len);
```

Comment:
Perhaps we can avoid this copy by using `std::string`?

https://github.com/cloudera/Impala/blob/cdh5-trunk/be/src/rpc/thrift-util.h#L85